src/test/java/com/doctusoft/math/TestExponentialIntDelays.java

```diff
@@ -0,0 +1,139 @@
+package com.doctusoft.math;
+
+import com.doctusoft.java.AnException;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.*;
+
+import static com.doctusoft.java.LambdAssert.assertThrows;
+import static com.doctusoft.math.ExponentialDelays.ints;
+import static com.doctusoft.math.ExponentialDelays.intsFrom;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.fail;
+
+public class TestExponentialIntDelays {
+    
+    @Test
+    public void defaultDelays() {
+        assertFirstDelays(ints().build(), 1, 2, 4, 8, 16, 32, 64);
+    }
+    
+    @Test
+    public void delaysWithInitialValue() {
+        assertFirstDelays(intsFrom(3).build(), 3, 6, 12, 24, 48, 96);
+        assertThrows(() -> intsFrom(0), AnException.of(IllegalArgumentException.class));
+        assertThrows(() -> intsFrom(-1), AnException.of(IllegalArgumentException.class));
+        int randomNegative = -1 * new Random().nextInt(Integer.MAX_VALUE);
```

Comment:
I don't like randomness in the tests. They can easily result in flaky tests.
If the assertion fails a line below you won't even know which number caused the problem.